src/components/last-planner/last-planner-view.tsx

```diff
@@ -0,0 +1,258 @@
+import { Task, TaskStatus, User } from "generated/client";
+import { useListJobPositionsQuery, useListTasksQuery, useListUsersQuery } from "hooks/api-queries";
+import { useApi } from "hooks/use-api";
+import { Fragment, useMemo } from "react";
+import { useTranslation } from "react-i18next";
+import { TaskWithInterval } from "types";
+import { getContrastForegroundColor, hexFromString } from "utils";
+import { splitIntervalByDuration } from "utils/date-time-utils";
+import {
+  distributeOverlappingTasksToRows,
+  renderTaskRows,
+  getTimelineIntervalByTasks,
+  groupTasksByOverlap,
+  mapTasksAndUsersByUserId,
+  sortTasksByStartTime,
+} from "utils/last-planner-utils";
+import { TaskRowCell } from "./task-row-cell";
+
+const LastPlannerTableWrapper = styled("div")(({ theme }) => ({
+  position: "relative",
+  paddingBottom: theme.spacing(2),
+  "& table": {
+    width: "100%",
+    borderCollapse: "separate",
+    borderSpacing: 0,
+    "& th": {
+      borderTop: `1px solid ${theme.palette.grey[300]}`,
+      borderBottom: `1px solid ${theme.palette.grey[300]}`,
+      borderRight: `1px solid ${theme.palette.grey[300]}`,
+    },
+    "& td": {
+      borderBottom: `1px solid ${theme.palette.grey[300]}`,
+      borderRight: `1px solid ${theme.palette.grey[300]}`,
+    },
+    "& th:first-of-type, & td:first-of-type": {
+      borderLeft: `1px solid ${theme.palette.grey[300]}`,
+    },
+  },
+}));
+
+const FixedHeightTableRow = styled("tr")({ height: 50 });
+
+type StickyTableCellProps = {
+  top?: number;
+  left?: number;
+  colSpan?: number;
+  constrainWidth?: boolean;
+};
+
+const StickyTableCell = styled("td", {
+  shouldForwardProp: (prop) =>
+    prop !== "top" && prop !== "left" && prop !== "noLeftBorder" && prop !== "constrainWidth",
+})<StickyTableCellProps>(({ top, left, colSpan = 1, constrainWidth = false }) => ({
+  position: "sticky",
+  top: top,
+  left: left,
+  backgroundColor: "white",
+  minWidth: 40,
+  maxWidth: constrainWidth ? 40 * colSpan : undefined,
+  verticalAlign: "middle",
+  overflow: "hidden",
+  textOverflow: "ellipsis",
+  zIndex: 1,
+}));
+
+type Props = {
+  projectId: string;
+  editMode?: boolean;
+};
+
+const LastPlannerView = ({ projectId, editMode }: Props) => {
+  const navigate = useNavigate({ from: "/projects/$projectId/tasks" });
+  const { t } = useTranslation();
+  const { tasksApi } = useApi();
+  const queryClient = useQueryClient();
+
+  const listTasksQuery = useListTasksQuery({ projectId });
+  const tasks = listTasksQuery.data ?? [];
+  const listProjectUsersQuery = useListUsersQuery({ projectId });
+  const users = listProjectUsersQuery.data?.users ?? [];
+  const jobPositionsQuery = useListJobPositionsQuery({ max: 9999 });
+  const jobPositions = jobPositionsQuery.data?.jobPositions ?? [];
+
+  const updateTaskMutation = useMutation({
+    mutationFn: (task: Task) => tasksApi.updateTask({ projectId: projectId, taskId: task.id as string, task: task }),
```

Comment:
Are you sure this happens even when you wait a bit? Currently we are updating the task status eagerly, so even though it updates "instantly" in the UI, it actually takes a bit to get the updated values from API. If between this time, you open the task, it might seem it did not update.

If this is the case, we could always add the eager update also to findTask query for the task currently being updated. This would mean a bit more work, but if you see it is needed, I will add that.

On the other hand, if the problem is not related to the eager update, then I'll investigate a bit more.

Reply:
I waited 10 seconds and still found it was as Daniil described. So it might be worth further investigation.

Couple of other things i noticed while looking at this:
-The add new task button does not seem to be working on the last planner screen.
-Can't update the status unless there is a dependent user- perhaps this is by design?
-Related to the above I notice we can't remove the dependent users to a state of no user- maybe this is fine?

Reply:
Okay, then I'll find out the cause and fix that.

- Add new task button is just for reference for now, it will come as a separate feature during the tasks list view. I have actually already started that.
- That is an overlook by me, I'll add some default icon when we don't have a dependent user.
- This is probably fine for now. I believe that if we _do_ have a dependent user at some point, it probably is not going to be removed alltogether. I also _hope_ that the end user would always place the dependent user, as why would a task be done if someone does not need the results of it 😁